miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultProjectService.java

```diff
@@ -160,6 +162,24 @@ public ProjectOverview getProjectOverviewById(long overviewId) throws IOExceptio
     return authorizationManager.filterUnreadable(projectStore.listAllWithLimit(limit));
   }
 
+  @Override
+  public Collection<Project> listAllProjectsByShortname() throws IOException {
+    List<Project> sortedProjects = (ArrayList<Project>) projectStore.listAll();
+
+    Collections.sort(sortedProjects,
+        /**
+         * Uses String.compareTo to alphabetically sort Projects by shortname
+         */
+        new Comparator<Project>() {
+          @Override
+          public int compare(Project p1, Project p2) {
+            return p1.getShortName().compareTo(p2.getShortName());
+          }
+        });
+
+    return authorizationManager.filterUnreadable(sortedProjects);
+  }
+
   private Collection<Group> loadManagedGroups(Collection<Group> original) throws IOException {
     if (original == null)
       return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)